control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -389,6 +389,19 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 			}
 			return err
 		}
+
+		if aws.StringValue(output.VpcEndpoints[0].ServiceName) != awsEndpointService.Status.EndpointServiceName {
+			log.Info("endpoint links to wrong endpointservice, deleting...", "VPCEndpointServiceName", aws.StringValue(output.VpcEndpoints[0].ServiceName), "AWSEndpointServiceName", awsEndpointService.Status.EndpointServiceName)
+			if _, err := ec2Client.DeleteVpcEndpointsWithContext(ctx, &ec2.DeleteVpcEndpointsInput{
+				VpcEndpointIds: []*string{output.VpcEndpoints[0].VpcEndpointId},
+			}); err != nil {
+				return fmt.Errorf("error deleting AWSEndpoint: %w", err)
+			}
+
+			// Once the VPC Endpoint is deleted, we need to send an error in order to reexecute the reconcilliation
+			return fmt.Errorf("conflicting VPCEndpoint found %s, reconciling...", aws.StringValue(output.VpcEndpoints[0].ServiceName))
```

Comment:
fwiw to be semantically concise we should add this function to return `ctrl.Result, err` and this should return `requeue true, nil`, I'm fine with current implementation for simplicity.

Reply:
Make sense, I think we can add it as a task for the controller refactor